### PR TITLE
(enh) Aider-Bench: make resumable with skip_num arg

### PR DESCRIPTION
**Pull request by tobitege**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Make Aider-Bench run resumable with optional `SKIP_NUM` env flag

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

- added optional `SKIP_NUM` env flag (int) which specifies the number of instances in the dataset to be skipped first, thus offering to resume a bench
- in `shared.py`, the `prepare_dataset` got a new optional parameter
- in that method, if a `skip_num` was given, it will combine this with the `eval_n_limit` number, so a run could skip the first 42 incidents and then run 10 instances
- temporarily changed links for huggingface evaluations to use the old repo name as the new one while the new one is in the works

### CODE REVIEW

File: evaluation/utils/shared.py [hunk 1]
```diff
@@ -181,34 +181,43 @@ def prepare_dataset(
...
     eval_ids: list[str] | None = None,
+    start_id: int | None = None,
 ):
     assert (
         'instance_id' in dataset.columns
     ), "Expected 'instance_id' column in the dataset. You should define your own unique identifier for each instance and use it as the 'instance_id' column."
     id_column = 'instance_id'
     logger.info(f'Writing evaluation output to {output_file}')
-    finished_ids = set()
+    finished_ids: set[int] = set()
```
Comment (xingyaoww) [inline evaluation/utils/shared.py:191]: Actually, the ID is not strictly INT - in fact, they are mostly string across all the evals, which makes start_id harder to support?

Comment (tobitege) [reply evaluation/utils/shared.py:191]: Oops, I'll change it to str then of course.

**Comment by tobitege** [comment]
@xingyaoww I've switched from `start_id` to `skip_num` instead, meaning the first `skip_num` entries from the dataset are skipped.
This removes the issues related to trying to use the main id as an int completely and can still be useful.
Changed documentation accordingly.

File: evaluation/README.md [hunk 2]
```diff
@@ -3,12 +3,14 @@
...
 To better organize the evaluation folder, we should follow the rules below:
-  - Each subfolder contains a specific benchmark or experiment. For example, `evaluation/swe_bench` should contain
+
+- Each subfolder contains a specific benchmark or experiment. For example, `evaluation/swe_bench` should contain
 all the preprocessing/evaluation/analysis scripts.
-  - Raw data and experimental records should not be stored within this repo.
-    - For model outputs, they should be stored at [this huggingface space](https://huggingface.co/spaces/OpenHands/evaluation) for visualization.
-  - Important data files of manageable size and analysis scripts (e.g., jupyter notebooks) can be directly uploaded to this repo.
+- Raw data and experimental records should not be stored within this repo.
+- For model outputs, they should be stored at [this huggingface space](https://huggingface.co/spaces/OpenHands/evaluation) for visualization.
```
Comment (xingyaoww) [inline evaluation/README.md:12]: We should use OpenDevin/evaluation as link for now until i fixed it :(

**Review by xingyaoww — APPROVED** [review]
LGTM! Thanks!!

File: evaluation/README.md (hunk 2 above)
Comment (tobitege) [reply evaluation/README.md:12]: Ohh, yes, that was a timely overlap now lol

File: evaluation/utils/shared.py (hunk 1 above)
Comment (tobitege) [reply evaluation/utils/shared.py]: Fixed

File: evaluation/README.md (hunk 2 above)
Comment (tobitege) [reply evaluation/README.md]: Fixed in several files

**Pull request merged**